### DISCUSSION TITLE
FR: Show Status Indicators for GraphQL Schema's

### DISCUSSION THREAD

**Discussion by michtio** [craftcms/cms#9031] [Ideas]
### Description

In GraphQL > Schemas you can create your own schemas and you have the Public Schema.

Currently the view looks as follows in the overview:

![Screenshot 2021-04-29 at 19 29 26](https://user-images.githubusercontent.com/5818021/116593281-972d1700-a929-11eb-86b0-86fc57a3102f.png)

There is no way to tell if a Schema is enabled or disabled. 
It would be of great help to show in the overview a red/green dot in front of the schemas to show if the schema is disabled/enabled.

**Comment by brandonkelly** [comment]
Schemas don’t have a status directly – just their tokens. I could see us adding a “Tokens” column to the index though, which lists out any tokens using each schema, and those could show status indicators.

**Comment by michtio** [comment]
@brandonkelly I get that schemas don't have a status directly, which is half/half, because the public one does, as seen in the below screenshot. 

![Screenshot 2021-05-03 at 10 38 06](https://user-images.githubusercontent.com/5818021/116856553-02b20580-ac04-11eb-8074-a0945e51a441.png)

It would be nice to see at least a status indicator for this one. 
But if we have enabled/disabled on the public schema, doesn't it make sense to just port that functionality over to each and every schema as well, including a status indicator? 

I can see a lot of possibilities adding the enabled/disabled including expiry date for tokens.
Eg a subscription form / mutation through graphQL to win something... where you can set an expiry, and create a schema just for that.

When the expiry dates gets hit, the schema gets disabled, FE wise you can code against it to give an error message that matches the expiry/no more info gotten.

**Comment by andris-sevcenko** [comment]
> I can see a lot of possibilities adding the enabled/disabled including expiry date for tokens.
Eg a subscription form / mutation through graphQL to win something... where you can set an expiry, and create a schema just for that.

Tokens already have those settings. It's schemas, that do not.

The awkwardness for public schema here comes from the fact that a public schema is used whenever no token is used. However, it also should be possible to disable/expire a public schema. So, the decision was made for public schemas to display those settings on a schema, as there's no token for public schema to configure.

This is also what makes it tricky just displaying that status on the schema list, because that status exists for a token, not schema.

**Comment by michtio** [comment]
Ah that's right @andris-sevcenko , I typed that expiry / enable/disable possibility when I wasn't decently awake yet it seems 😬
